exercicios/para-casa/sistema_de_alunas.SamanthaTeng.py

```diff
@@ -0,0 +1,176 @@
+        cod_opcao = obter_opcao()
+        
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas()
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+
+def incluir_nova_aluna():
+    print("Insira os seguintes dados: ")
+    nome = input("Nome da aluna: ") 
+    sobrenome = input ("Sobrenome: ")
+    turma = input("Turma da aluna: ") 
+    lista_presenca = obter_presenca() 
+    nota_participacao = float(input("Participação da aluna: "))
+    notas = obter_notas() 
+    salvar_dados_aluna(nome,sobrenome, turma, notas, lista_presenca, nota_participacao) 
+    return nome
+
+
+def obter_presenca():
+    quantidade_aulas = input("Quantidade de aulas: ") 
+    aulas = [] 
+    
+    for contador in range(int(quantidade_aulas)): 
+        
+        while True:    
+            entrada = input(f"Insira a presença da aula #{contador + 1}: ") 
+            try: 
+                presenca = eval(entrada) 
+                aulas.append(presenca) 
+                break 
+            except NameError: 
+                print("Entrada inválida. Por favor, insira True ou False.")
+   
+    return aulas
+    
+def obter_notas():
+
+    quantidade_notas = input("Quantidade de notas: ") 
+    notas = [] 
+    
+    for contador in range(int(quantidade_notas)): 
+       
+        while True: 
+            entrada = input(f"Insira a nota #{contador + 1}: ") 
+            try: 
+                nota = float(entrada) 
+                notas.append(nota) 
+                break 
+            except ValueError:
+                print("Entrada inválida. Por favor, insira um número válido.")
+                
+    
+    return notas
+
+def salvar_dados_aluna (nome,sobrenome, turma, notas, lista_presenca, nota_participacao):
+    chave = (nome) 
+    dataset[chave] = { 
+        "Sobrenome": sobrenome,
```

Comment:
O sobrenome deve estar dentro da chave na tupla. Podemos apenas incluir dentro da tupla.

**Sugestão:**
```suggestion
    chave = (nome, sobrenome) 
    dataset[chave] = { 
```